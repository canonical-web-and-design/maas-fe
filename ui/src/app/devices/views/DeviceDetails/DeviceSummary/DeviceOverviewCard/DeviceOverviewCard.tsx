import { Card, Spinner } from "@canonical/react-components";

import LabelledList from "app/base/components/LabelledList";
import type { Device } from "app/store/device/types";
import { isDeviceDetails } from "app/store/device/utils";

type Props = {
  device: Device;
};

const DeviceOverviewCard = ({ device }: Props): JSX.Element => {
  const {
    domain: { name: domainName },
    owner,
    tags,
    zone: { name: zoneName },
  } = device;

  return (
    <Card>
      <h4 className="p-muted-heading u-sv1">Overview</h4>
      <hr />
      <LabelledList
        className="u-no-margin--bottom"
        items={[
          {
            label: "Owner",
            value: owner || "—",
          },
          {
            label: "Domain",
            value: domainName || "—",
          },
          {
            label: "Zone",
            value: zoneName || "—",
          },
          {
            label: "Note",
            value: isDeviceDetails(device) ? (
              device.description || "—"
            ) : (
              <Spinner data-testid="loading-note" />
            ),
          },
          {
            label: "Tags",
            value: tags.join(", ") || "—",
          },
        ]}
      />
    </Card>
  );
};

export default DeviceOverviewCard;
