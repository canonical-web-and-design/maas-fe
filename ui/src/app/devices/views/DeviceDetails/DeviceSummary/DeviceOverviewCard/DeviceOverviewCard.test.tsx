import { mount, shallow } from "enzyme";

import DeviceOverviewCard from "./DeviceOverviewCard";

import {
  device as deviceFactory,
  deviceDetails as deviceDetailsFactory,
} from "testing/factories";

describe("DeviceOverviewCard", () => {
  it("renders", () => {
    const device = deviceDetailsFactory({
      description: "description",
      domain: { id: 1, name: "domain" },
      owner: "Owner",
      tags: ["tag1", "tag2"],
      zone: { id: 1, name: "zone" },
    });
    const wrapper = shallow(<DeviceOverviewCard device={device} />);

    expect(wrapper.children()).toMatchSnapshot();
  });

  it("shows a spinner for the note if not device details", () => {
    const device = deviceFactory();
    const wrapper = mount(<DeviceOverviewCard device={device} />);

    expect(wrapper.find("[data-testid='loading-note']").exists()).toBe(true);
  });
});
