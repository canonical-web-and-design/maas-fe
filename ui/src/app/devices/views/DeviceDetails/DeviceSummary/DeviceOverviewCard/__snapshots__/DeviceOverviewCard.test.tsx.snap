// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceOverviewCard renders 1`] = `
Array [
  <h4
    className="p-muted-heading u-sv1"
  >
    Overview
  </h4>,
  <hr />,
  <LabelledList
    className="u-no-margin--bottom"
    items={
      Array [
        Object {
          "label": "Owner",
          "value": "Owner",
        },
        Object {
          "label": "Domain",
          "value": "domain",
        },
        Object {
          "label": "Zone",
          "value": "zone",
        },
        Object {
          "label": "Note",
          "value": "description",
        },
        Object {
          "label": "Tags",
          "value": "tag1, tag2",
        },
      ]
    }
  />,
]
`;
